src/NodeApi/JSFunction.cs

```diff
@@ -0,0 +1,357 @@
+
+    /// <summary>
+    /// Creates a new named JS function takes two arguments and returns a value.
+    /// </summary>
+    public JSFunction(string name, Func<JSValue, JSValue, JSValue> callback)
+        : this(JSValue.CreateFunction(name, (args) => callback(args[0], args[1])))
+    {
+    }
+
+    /// <summary>
+    /// Creates a new named JS function takes three arguments and returns a value.
+    /// </summary>
+    public JSFunction(string name, Func<JSValue, JSValue, JSValue, JSValue> callback)
+        : this(JSValue.CreateFunction(name, (args) => callback(args[0], args[1], args[2])))
+    {
+    }
+
+    /// <summary>
+    /// Creates a new named JS function takes four arguments and returns a value.
+    /// </summary>
+    public JSFunction(string name, Func<JSValue, JSValue, JSValue, JSValue, JSValue> callback)
+        : this(JSValue.CreateFunction(name, (args) => callback(args[0], args[1], args[2], args[3])))
+    {
+    }
+
+    /// <summary>
+    /// Creates a new named JS function takes five arguments and returns a value.
+    /// </summary>
+    public JSFunction(
+        string name, Func<JSValue, JSValue, JSValue, JSValue, JSValue, JSValue> callback)
+        : this(JSValue.CreateFunction(name, (args) =>
+            callback(args[0], args[1], args[2], args[3], args[4])))
+    {
+    }
+
+    /// <summary>
+    /// Gets the name of the function, or an empty string if the function is unnamed.
+    /// </summary>
+    public string Name => (string)_value["name"];
+
+    /// <summary>
+    /// Calls the function with the specified `this` value.
+    /// </summary>
+    public JSValue Apply(JSValue thisArg)
+    {
+        return _value.CallMethod("apply", thisArg);
+    }
+
+    /// <summary>
+    /// Calls the function with the specified `this` value and arguments.
+    /// </summary>
+    public JSValue Apply(JSValue thisArg, JSArray args)
+    {
+        return _value.CallMethod("apply", thisArg, args);
+    }
+
+    /// <summary>
+    /// Calls the function with the specified `this` value and arguments.
+    /// </summary>
+    public JSValue Apply(JSValue thisArg, JSValue[] args)
+    {
+        return _value.CallMethod("apply", thisArg, new JSArray(args));
+    }
+
+    /// <summary>
+    /// Creates a new function that when called has the specified `this` value, and
+    /// optionally the specified sequence of arguments preceding any provided when the
+    /// new function is called.
+    /// </summary>
+    public JSFunction Bind(JSValue thisArg, params JSValue[] args)
+    {
+        if (args.Length == 0)
+        {
+            return (JSFunction)_value.CallMethod("bind", thisArg);
+        }
+        else
+        {
+            JSValue[] thisAndArgs = new JSValue[1 + args.Length];
+            thisAndArgs[0] = thisArg;
+            args.CopyTo(thisAndArgs, 1);
+            return (JSFunction)_value.CallMethod("bind", thisAndArgs);
+        }
+    }
+
+    public JSValue Call() => _value.Call(thisArg: default);
```

Comment:
>Call() [](http://example.com/codeflow?start=19&length=6)

It has a different semantic from the JSValue.Call.
Should we keep them aligned and have a different name when function is called as standalone?
JavaScript Function object has `apply` and `call` methods. The JSFunction.Apply corresponds to Function.protype.apply. I would expect the JSFunction.Call would do the same in regard to the Function.protype.call.
The new candidate names are StaticCall and CallAsStatic.

Reply:
Yes, good point.